week3/Program.cs

```diff
@@ -0,0 +1,135 @@
+var builder = WebApplication.CreateBuilder(args);
+var app = builder.Build();
+
+app.MapGet("/", () => "Hello World!");
+
+app.MapGet("/class-exer1", () =>
+{
+    var temperature = new Temperature(300);
+    Console.WriteLine($"{temperature.Celsius} Celsius is {temperature.ConvertToFahrenheit()} Fahrenheit");
+});
+
+app.MapGet("/class-exer2", () =>
+{
+    var amount = 100;
+    var exchangeRate = new ExchangeRate("EUR", "DKK");
+    exchangeRate._rate = 7.5m;
+    Console.WriteLine($"{amount} {exchangeRate.FromCurrency} is {exchangeRate.Calculate(amount)} {exchangeRate.ToCurrency}");
+
+});
+
+app.MapGet("/hw-exer1", () =>
+{
+    var account = new Account(100.56m);
+    account.Deposit(100m);
+    Console.WriteLine($"Account balance is {account.Balance}");
+    account.Withdraw(20);
+    Console.WriteLine($"Account balance is {account.Balance}");
+    //account.Withdraw(200);
+});
+
+app.MapGet("/hw-exer2", () =>
+{
+    var myCow = new Cow();
+    MakeSound(myCow);
+    var myCat = new Cat();
+    MakeSound(myCat);
+    var myDog = new Dog();
+    MakeSound(myDog);
+});
+
+void MakeSound(IAnimal animal)
+{
+    Console.WriteLine($"{animal.Name} says {animal.Sound}");
+};
+
+app.Run();
+
+// class- exercise 1
+public class Temperature
+{
+    public decimal Celsius { get; private set; }
+    public Temperature(decimal celsius)
+    {
+        Celsius = celsius;
+        if (Celsius < 273.15m)

```

Comment:
this check should be before the assignment of the Celsius property (Celsius = celsius;)
Otherwise the temperature will still be assigned with the incorrect value